.github/workflows/pr.yaml

```diff
@@ -55,3 +57,19 @@ jobs:
       arch: "amd64"
       container_image: "rapidsai/ci:latest"
       run_script: "ci/build_docs.sh"
+  wheel-build:
+    needs: checks
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-pure-build.yml@cuda-120
+    with:
+      build_type: pull-request
+      package-name: cuxfilter
+      package-dir: python/
+  wheel-tests:
+    needs: wheel-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-pure-test.yml@cuda-120
+    with:
+      build_type: pull-request
+      package-name: cuxfilter
+      test-unittest: "python -m pytest -n 8 ./python/cuxfilter/tests"
```

Comment:
You'll need to borrow some logic from here for dask-cuda: https://github.com/rapidsai/cudf/blob/1899fb1990eae693bd74f92fa4a3f6a217889210/.github/workflows/pr.yaml#L132-L133

dask-cuda has to install from source, since the `23.8.*a*` packages on PyPI are pre-release / alphas. We can't really use `pip install ... --pre` because it uses pre-release versions of all packages.

I'm not sure if you need _just_ dask-cuda or also dask/distributed.

Also note that you'll need to change `update-version.sh` to accomodate this. https://github.com/rapidsai/cudf/blob/1899fb1990eae693bd74f92fa4a3f6a217889210/ci/release/update-version.sh#L112

All of dask/distributed/dask-cuda:
```suggestion
      # Test against latest dask/distributed/dask-cuda.
      test-before: "python -m pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.08"
      test-unittest: "python -m pytest -n 8 ./python/cuxfilter/tests"
```

Only dask-cuda:
```suggestion
      # Test against latest dask-cuda.
      test-before: "python -m pip install git+https://github.com/rapidsai/dask-cuda.git@branch-23.08"
      test-unittest: "python -m pytest -n 8 ./python/cuxfilter/tests"
```

Reply:
ah okay got it. It would just be dask-cuda at this point. Thanks again!

@bdice I am assuming this is a temporary fix, until dask-cuda has a pip release (23.10.*?), right?

Reply:
Actually @bdice, we don't need dask_cuda as a direct dependency, since it's only used in the dask_cudf based notebooks. So I removed the direct dependency, and added it to `notebooks` dependency only

Reply:
Great that the dependency could be moved.

To answer your question about whether this would be a temporary fix, it isn't temporary. The dask-cuda wheels used in CI are always pre-release alphas. We are always testing RAPIDS packages on `branch-XX.YY` against pre-releases of dask-cuda `XX.YY` to ensure that when the releases are made, everything works together as intended.

Reply:
got it, good to know. Thanks again for the thorough review. Let me know if this looks good to merge.